Content.Server/Mining/MiningSystem.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Content.Server.Mining.Components;
+using Content.Server.Stack;
+using Content.Shared.Destructible;
+using Content.Shared.Mining;
+using Content.Shared.Prototypes;
+using Content.Shared.Random;
+using Content.Shared.Random.Helpers;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+
+namespace Content.Server.Mining;
+
+/// <summary>
+/// This handles creating ores when the entity is destroyed.
+/// </summary>
+public sealed class MiningSystem : EntitySystem
+{
+    [Dependency] private readonly IPrototypeManager _proto = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly StackSystem _stack = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        base.Initialize();
+        SubscribeLocalEvent<OreVeinComponent, MapInitEvent>(OnMapInit);
+        SubscribeLocalEvent<OreVeinComponent, DestructionEventArgs>(OnDestruction);
+    }
+
+    private void OnDestruction(EntityUid uid, OreVeinComponent component, DestructionEventArgs args)
+    {
+        if (component.CurrentOre == null)
+            return;
+
+        if (!_proto.TryIndex<OrePrototype>(component.CurrentOre, out var proto))
+            return;
```

Comment:
Does TryIndex take a nullable? Also you should log if this fails because it shouldn't be happening.